### PR TITLE
Refactor lily58

### PR DESCRIPTION
**Pull request by waffle87**
## Description

* Move all revisions to data driven
* Create keyboard level `info.json` with things that are shared across all revisions (matrix pins, layout, split communication, diode direction)
* Keyboard level encoder & OLED functionality (and removed from keymaps where applicable)
* Remove use of `sprintf` in keylogger
* Move `r2g` OLED configuration (where it differs) to `mb_via` keymap (@e11i0t23)
* Other general cleanup

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/lily58/info.json [hunk 1]
```diff
@@ -0,0 +1,83 @@
+{
+  "feautres": {
```
Comment (frobiac) [inline keyboards/lily58/info.json:2]: Typo :-)

Comment (waffle87) [reply keyboards/lily58/info.json:2]: aha! tks.

File: keyboards/lily58/r2g/r2g.c [hunk 2]
```diff
@@ -1,169 +0,0 @@
...
-    return false;
-}
-
-bool process_record_kb(uint16_t keycode, keyrecord_t *record) {
-  if (record->event.pressed) {
-    set_keylog_r2g(keycode, record);
-  }
-  return process_record_user(keycode, record);
-}
-#endif // OLED_ENABLE
```
Comment (daskygit) [inline keyboards/lily58/r2g/r2g.c:169]: Why have you moved this?

Comment (waffle87) [reply keyboards/lily58/r2g/r2g.c:169]: Would conflict with implementation in `lily58.c`, and figured this was better than copy/pasting same implementation in `glow_enc`, `light`, & `rev1`.

Comment (daskygit) [reply keyboards/lily58/r2g/r2g.c:169]: Those revisions didn't have any keyboard level implementation for OLED control, how are you checking the binaries produced are the same if you're editing the default behaviour?

**Pull request closed without merging**